Makefile

```diff
@@ -1,17 +1,38 @@
+.PHONY: build fmt check-fmt lint lint-fix codespell
+
+DESTDIR ?=
 
 ALL_SRC_FILES = 'find . -name "*.[ch]" ! -path "./src/libhirte/ini/ini.[ch]" ! -path "./src/libhirte/hashmap/*" ! -path "./src/*/test/**" ! -path "./builddir/**" -print0'
 
+build:
+	meson setup builddir
+	meson compile -C builddir
+
+test: build
+	meson setup builddir
+	meson compile -C builddir
+	meson test -C builddir
+
+install: build
+	meson install -C builddir --destdir "$(DESTDIR)"
+
 fmt:
 	eval $(ALL_SRC_FILES) | xargs -0 -I {} clang-format -i --sort-includes {}
 
 check-fmt:
 	eval $(ALL_SRC_FILES) | xargs -0 -I {} clang-format  --dry-run -Werror {}
 
-lint: 
+lint:
 	eval $(ALL_SRC_FILES) | xargs -0 -I {} clang-tidy --quiet {} -- -I src/ -D_GNU_SOURCE
 
-lint-fix: 
+lint-fix:
 	eval $(ALL_SRC_FILES) | xargs -0 -I {} clang-tidy --quiet {} --fix -- -I src/ -D_GNU_SOURCE
 
 codespell:
 	codespell -S Makefile,imgtype,copy,AUTHORS,bin,.git,CHANGELOG.md,changelog.txt,.cirrus.yml,"*.xz,*.gz,*.tar,*.tgz,*ico,*.png,*.1,*.5,*.orig,*.rej,*.xml" -L keypair,flate,uint,iff,od,ERRO -w
+
+clean:
+	rm -rf builddir
+	meson setup builddir
```

Comment:
Why do you need to run the `setup` command during `clean`? Won't it run for `build`?

Reply:
It is a chicken and egg, since builddir already deleted, 
Maybe using this 
`meson setup --wipe builddir`

Could be better,